lib/system.nim

```diff
@@ -2580,7 +2580,7 @@ template once*(body: untyped): untyped =
 
 {.pop.} # warning[GcMem]: off, warning[Uninit]: off
 
-proc substr*(s: string, first, last: int): string =
+proc substr*(s: openArray[char], first, last: int): string =
```

Comment:
But thanks to `toOpenArray` the slice `first..last` idea is already part of the `openArray[char]` type. This is a very weird interface.

Reply:
`openArray[char][a..b]` returns a `seq[char]` and not a `string`,  so in the cases `substr` is used one will likely need an equivalent regardless, perhaps it should be named `toString` instead for the `openarray` variant?

Reply:
I didn't say `openArray[char][a..b]`, I said `toOpenArray`.